enterprise/internal/scim/user.go

```diff
@@ -70,38 +76,75 @@ func (h *UserResourceHandler) Get(r *http.Request, idStr string) (scim.Resource,
 // An empty list of resources will be represented as `null` in the JSON response if `nil` is assigned to the
 // Page.Resources. Otherwise, is an empty slice is assigned, an empty list will be represented as `[]`.
 func (h *UserResourceHandler) GetAll(r *http.Request, params scim.ListRequestParams) (scim.Page, error) {
+	var totalCount int
+	var resources []scim.Resource
+	var err error
+
+	if params.Filter == nil {
+		totalCount, resources, err = h.getAllFromDB(r, params.StartIndex, &params.Count)
+	} else {
+		// Fetch all resources from the GB and then filter them here.
```

Comment:
```suggestion
		// Fetch all resources from the DB and then filter them here.
```

I assume this is supposed to read DB?